views/admin/object.erb

```diff
@@ -55,6 +55,8 @@
 </tbody>
 </table>
 
+<%== part("extras/#{@klass.name}", obj: @obj) if File.exist?("views/admin/extras/#{@klass.name}.erb") %>
```

Comment:
Instead of checking for file existence in every render, can you instead add a hash to the app with a model name key and true value for the models that have extra info, and check that hash?